launcher/modplatform/flame/FlameInstanceCreationTask.cpp

```diff
@@ -520,3 +506,110 @@ void FlameCreationTask::setupDownloadJob(QEventLoop& loop)
     setStatus(tr("Downloading mods..."));
     m_files_job->start();
 }
+
+/// @brief copy the matched blocked mods to the instance staging area
+/// @param blocked_mods list of the blocked mods and their matched paths
+void FlameCreationTask::copyBlockedMods(QList<BlockedMod> const& blocked_mods)
+{
+    setStatus(tr("Copying Blocked Mods..."));
+    setAbortable(false);
+    int i = 0;
+    int total = blocked_mods.length();
+    setProgress(i, total);
+    for (auto const& mod : blocked_mods) {
+        if (!mod.matched) {
+            qDebug() << mod.name << "was not matched to a local file, skipping copy";
+            continue;
+        }
+
+        auto destPath = FS::PathCombine(m_stagingPath, "minecraft", mod.targetFolder, mod.name);
+
+        setStatus(tr("Copying Blocked Mods (%1 out of %2 are done)").arg(QString::number(i), QString::number(total)));
+
+        qDebug() << "Will try to copy" << mod.localPath << "to" << destPath;
+
+        if (!FS::copy(mod.localPath, destPath)()) {
+            qDebug() << "Copy of" << mod.localPath << "to" << destPath << "Failed";
+        }
+
+        i++;
+        setProgress(i, total);
+    }
+
+    setAbortable(true);
+}
+
+bool moveFile(QString src, QString dst)
+{
+    if (!FS::copy(src, dst)()) {  // copy
+        qDebug() << "Copy of" << src << "to" << dst << "failed!";
+        return false;
+    } else {
+        if (!FS::deletePath(src)) {  // remove original
+            qDebug() << "Deletion of" << src << "failed!";
+            return false;
+        };
+    }
+    return true;
```

Comment:
shouldn't this just be a function in `FileSystem.h` using [`std::filesystem::rename`](https://en.cppreference.com/w/cpp/filesystem/rename)? Since it follows the [POSIX spec](https://pubs.opengroup.org/onlinepubs/9699919799/functions/renameat.html), in a failure it will not change the src file, so it retains the behavior you seem to be aiming here.

Reply:
... WOW, this is peak not written here syndrome. why do I do this to myself?